v2/tools/generator/internal/functions/kubernetes_admissions_validations.go

```diff
@@ -77,3 +85,59 @@ func validateResourceReferencesBody(codeGenerationContext *astmodel.CodeGenerati
 
 	return body
 }
+
+func validateImmutablePropertiesFunction(resourceFn *ResourceFunction, codeGenerationContext *astmodel.CodeGenerationContext, receiver astmodel.TypeName, methodName string) *dst.FuncDecl {
+
+	receiverIdent := resourceFn.IdFactory().CreateReceiver(receiver.Name())
+	receiverType := receiver.AsType(codeGenerationContext)
+
+	runtimePackage := codeGenerationContext.MustGetImportedPackageName(astmodel.APIMachineryRuntimeReference)
+
+	fn := &astbuilder.FuncDetails{
+		Name:          methodName,
+		ReceiverIdent: receiverIdent,
+		ReceiverType: &dst.StarExpr{
+			X: receiverType,
+		},
+		Returns: []*dst.Field{
+			{
+				Type: dst.NewIdent("error"),
+			},
+		},
+		Body: validateImmutablePropertiesFunctionBody(receiver, codeGenerationContext, receiverIdent),
+	}
+
+	fn.AddParameter("old", astbuilder.QualifiedTypeName(runtimePackage, "Object"))
+	fn.AddComments("validates all immutable properties")
+
+	return fn.DefineFunc()
+}
+
+// validateImmutablePropertiesFunctionBody helps generate the body of the validateImmutablePropertiesFunctionBody function:
+//  oldObj, ok := old.(*Receiver)
+//  if !ok {
+//		return nil
+//  }
+//
+// 	return genruntime.ValidateImmutableProperties(oldObj, <receiverIndent>)
+func validateImmutablePropertiesFunctionBody(receiver astmodel.TypeName, codeGenerationContext *astmodel.CodeGenerationContext, receiverIdent string) []dst.Stmt {
+
+	genRuntime := codeGenerationContext.MustGetImportedPackageName(astmodel.GenRuntimeReference)
+
+	obj := dst.NewIdent("oldObj")
+
+	cast := astbuilder.TypeAssert(obj, dst.NewIdent("old"), astbuilder.Dereference(receiver.AsType(codeGenerationContext)))
+	checkAssert := astbuilder.ReturnIfNotOk(astbuilder.Nil())
+
+	returnStmt := astbuilder.Returns(
```

Comment:
Suggest configuring this to have a blank line before the **return** statement.

Reply:
Added!